documentation/modules/oauth/con-oauth-client-config.adoc

```diff
@@ -0,0 +1,176 @@
+:_mod-docs-content-type: CONCEPT
+
+[id='con-oauth-client-config-{context}']
+= Client-side configuration
+
+[role="_abstract"]
+Kafka clients, including Strimzi-managed components and external applications, can authenticate to Kafka using OAuth 2.0 access tokens. 
+Clients obtain a token from an authorization server and present it to Kafka using the `OAUTHBEARER` SASL mechanism. 
+Kafka brokers validate the token using the OAuth configuration defined on their listeners.
+
+Client applications must use OAuth settings that match the broker configuration, including token type, issuer, and TLS trust configuration.
+
+This section describes:
+
+* OAuth requirements common to all Kafka clients
+* How Strimzi components authenticate as OAuth clients
+* How external Kafka applications configure `OAUTHBEARER` authentication
+* How clients obtain and refresh tokens
+
+NOTE: Advanced OAuth client settings are described in the {oauth-project}.
+
+== Common requirements for OAuth 2.0 clients
+
+Kafka clients that use OAuth-based authentication typically require:
+
+* The `OAUTHBEARER` SASL mechanism  
+* A token endpoint to obtain OAuth 2.0 access tokens  
+* A client ID and credentials registered with the authorization server  
+* TLS trust configuration for HTTPS calls to the authorization server  
+* A way to refresh tokens before they expire  
+* Truststore configuration that matches the Kafka broker listener
+
+Strimzi does not decide which OAuth 2.0 flow is used. 
+The chosen flow depends on the authorization server and the client application.
+
+== Configuring Strimzi-managed components
+
+Strimzi-managed components such as Kafka Connect, MirrorMaker 2, and HTTP Bridge authenticate as OAuth 2.0 clients when connecting to Kafka brokers. 
+These components use the `KafkaClientAuthenticationCustom` schema and configure OAuth by supplying SASL and JAAS settings.
+
+Each component requires:
+
+* A token endpoint URL  
+* A client ID and credentials stored in a Kubernetes secret  
+* TLS trust configuration for connecting to the authorization server  
+* `OAUTHBEARER` SASL settings to authenticate to Kafka brokers
+
+The following example shows a simplified OAuth configuration for the HTTP Bridge. 
+The same structure applies to Kafka Connect and MirrorMaker 2.
+
+[source,yaml]
+----
+apiVersion: kafka.strimzi.io/v1
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  # ...
+  authentication:
+    type: custom
+    sasl: true
+    config:
+      sasl.mechanism: OAUTHBEARER
+      sasl.login.callback.handler.class: io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler
+      sasl.jaas.config: |
+        org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required
+          oauth.token.endpoint.uri="https://<auth_server_address>/<path_to_token_endpoint>"
+          oauth.client.id="kafka-bridge"
+          oauth.client.secret="${strimzienv:OAUTH_CLIENT_SECRET}";
```

Comment:
Does this actually work? Can this be used and is automatically expanded into the env var value when KafkaBridge is started? If so, the same approach can be used when configuring the `oauth.client.secret` in the listener if the same kind of expansion takes place.

Reply:
It was based on the example in the proposal.
we have some docs on: [loading external values](https://strimzi.io/docs/operators/in-development/deploying#assembly-loading-config-with-providers-str)